core/src/main/scala/org/apache/spark/storage/MemoryStore.scala

```diff
@@ -418,9 +423,11 @@ private[spark] class MemoryStore(blockManager: BlockManager, memoryManager: Memo
         while (freedMemory < space && iterator.hasNext) {
           val pair = iterator.next()
           val blockId = pair.getKey
-          if (rddToAdd.isEmpty || rddToAdd != getRddId(blockId)) {
-            selectedBlocks += blockId
-            freedMemory += pair.getValue.size
+          if (blockIsEvictable(blockId)) {
+            if (blockManager.blockInfoManager.lockForWriting(blockId, false).isDefined) {
```

Comment:
name boolean parameter, also add a comment to explain why we don't want to block here.